File: Demos/Datasnap/Server/proxy/objectivec_ios81/Proxy/TJSONValue.m
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

#import "TJSONValue.h"


@implementation TJSONValue


- (NSString *) nullString{
	 return @"null";
}

-(id) getInternalObject{
	return nil;	
}

-(NSArray*) asJSONArray{
	return nil;
}

-(NSString *) asJSONString{
	return nil;
}
-(NSDictionary *) asJSONObject{
	return nil;
}
-(NSString *) toString{
	return [self nullString];

}
-(JSONValueType) getJSONValueType{
	return JSONNull;
}
@end

File: Demos/Datasnap/Server/proxy/java_android/com/embarcadero/javaandroid/TJSONValue.java
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

package com.embarcadero.javaandroid;

/**
 * Represents the ancestor class for the TJSON classes .
 */

public abstract class TJSONValue {

	/**
	 * A String that represents the JSON Null String that is "null"
	 */
	final protected String NullString = "null";

	/**
	 * Returns the specified internal JSONValue wrapped
	 * 
	 * @return the internal object
	 */
	public abstract Object getInternalObject();

	/**
	 * Returns the specified {@link JSONValueType} of the internal JSONValue
	 * wrapped
	 * 
	 * @return
	 */
	public abstract JSONValueType getJsonValueType();

	/**
	 * Returns the JSON String representation for this object
	 */
	public abstract String toString();
}
